setup.sh

```diff
@@ -0,0 +1,101 @@
+#!/bin/bash
+
+RED='\033[0;31m'
+NC='\033[0m'
+python_command="python"
+pip_command="pip"
+
+function checkPythonVersion () {
+    case "$($1 --version 2>&1)" in
+    *" 3."*)
+        return 1
+        ;;
+    *)
+        return 0
+        ;;
+    esac
+}
+
+function checkPIPVersion () {
+    case "$($1 --version 6>&1)" in
+    *" 3."*)
+        return 1
+        ;;
+    *)
+        return 0
+        ;;
+    esac
+}
+
+function main () {
+    if checkPythonVersion $python_command; then
+        python_command="python3"
+        if checkPythonVersion $python_command; then
+            echo "Python3 not found."
+            exit 1
+        fi
+        echo "Using python3 instead of python"
+    fi
+
+    if checkPIPVersion $pip_command; then
+        pip_command="pip3"
+        if checkPIPVersion $pip_command; then
+            echo "pip for Python3 not found."
+            exit 1
+        fi
+        echo "Using pip3 instead of pip"
+    fi
+
+    $pip_command install -r requirements.txt
+}
+
+function full () {
+    echo -e "Running the ${RED}full${NC} setup (including building the model)"
+    main
+    mkdir data
```

Comment:
should we use absolute path for everything? Just to be safe.

Reply:
Not really because when the Python code is ran, it is also relative to wherever this repo itself was cloned. We don't really want to absolute it into some config folder that might need some additional work to remove if we ever want to "uninstall" this project. Its safer to make it a relative path than absolute path in this case imo.

Reply:
I'm just worried about the possibility the script could be run from another directory or someday another script would calls this script for setup.